### PR TITLE
Added support for snowflake database in database package

### PR DESCRIPTION
**Pull request by utkarsharma2**
Closes: #333  and #339

### CODE REVIEW

File: src/astro/databases/snowflake.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+from astro.databases.base import BaseDatabase
+from astro.sql.tables import Table
+from astro.utils.dependencies import pandas_tools
+
+DEFAULT_CONN_ID = SnowflakeHook.default_conn_name
+
+
+class SnowflakeDatabase(BaseDatabase):
+    """
+    Handle interactions with Postgres databases. If this class is successful, we should not have any Postgres-specific
```
Comment (kaxil) [inline src/astro/databases/snowflake.py:15]: Docstrings needs to be updated to Snowflake :) -- currently it talks about Postgres

**Comment by kaxil** [comment]
@utkarsharma2 You will need to rebase :)

File: src/astro/databases/snowflake.py (hunk 1 above)
Comment (utkarsharma2) [reply src/astro/databases/snowflake.py:15]: Opps, done. :)

File: tests/databases/test_snowflake.py [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+
+
+@pytest.mark.integration
+@pytest.mark.parametrize(
+    "database_table_fixture",
+    [
+        {
+            "database": Database.SNOWFLAKE,
+            "table": Table(metadata=Metadata(schema=SCHEMA)),
+            "file": File(str(pathlib.Path(CWD.parent, "data/sample.csv"))),
```
[the remark this answers is not in the document]
Comment (utkarsharma2) [reply tests/databases/test_snowflake.py:268]: This fixture is populating the file in database and then we are exporting that table's data to a file in gcs.

**Review by kaxil — APPROVED** [review]

**Pull request merged**